src/storage/jni.rs

```diff
@@ -0,0 +1,320 @@
+    _class: JClass,
+    _storage: *mut Storage,
+)
+{
+    if !_storage.is_null(){
+        drop(unsafe { Box::from_raw(_storage) })
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeGet(
+    _env: JNIEnv,
+    _storage: JObject,
+    _cf: JObject,
+    _key: jbyteArray
+) -> jbyteArray
+{
+    let storage = unwrap_ptr::<Storage>(&_env, _storage);
+    let cf = unwrap_ptr::<ColumnFamily>(&_env, _cf);
+
+    let key = _env.convert_byte_array(_key)
+        .expect("Should be able to convert to Rust byte array");
+
+    if let Some(value) = storage.get_cf(cf, key.as_slice()){
+        _env.byte_array_from_slice(value.as_slice())
+            .expect("Should be able to convert Rust slice into jbytearray")
+    } else {
+        null_mut()
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeMultiGet(
+    _env: JNIEnv,
+    _storage: JObject,
+    _cf: JObject,
+    _keys: jobjectArray
+) -> jobject
+{
+    let storage = unwrap_ptr::<Storage>(&_env, _storage);
+    let cf = unwrap_ptr::<ColumnFamily>(&_env, _cf);
+    let keys = java_array_to_vec_byte(&_env, _keys);
+
+    let key_values = storage.multi_get_cf(
+        cf,
+        keys.iter().map(|k|k.as_slice()).collect::<Vec<_>>().as_slice()
+    );
+    map_to_java_map(&_env, &key_values)
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeIsEmpty(
+    _env: JNIEnv,
+    _storage: JObject,
+    _cf: JObject,
+) -> jboolean {
+    let storage = unwrap_ptr::<Storage>(&_env, _storage);
+    let cf = unwrap_ptr::<ColumnFamily>(&_env, _cf);
+
+    if let Ok(is_empty) = storage.is_empty_cf(cf){
+        is_empty as jboolean
+    } else {
+        JNI_TRUE
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeSetColumnFamily(
+    _env: JNIEnv,
+    _storage: JObject,
+    _cf_name: JString
+) -> jboolean {
+    let storage = unwrap_mut_ptr::<Storage>(&_env, _storage);
+
+    let cf_name = _env
+        .get_string(_cf_name)
+        .expect("Should be able to read jstring as Rust String");
+
+    if let Ok(()) = storage.set_column_family(cf_name.to_str().unwrap()){
+        JNI_TRUE
+    } else {
+        JNI_FALSE
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeGetColumnFamily(
+    _env: JNIEnv,
+    _storage: JObject,
+    _cf_name: JString
+) -> jobject
+{
+    let storage = unwrap_ptr::<Storage>(&_env, _storage);
+
+    let cf_name = _env
+        .get_string(_cf_name)
+        .expect("Should be able to read jstring as Rust String");
+
+    if let Some(cf_ref) = storage.get_column_family(cf_name.to_str().unwrap()){
+        let column_family_class = _env.find_class("com/horizen/common/ColumnFamily")
+            .expect("Should be able to find class ColumnFamily");
+        // Converting the cf_ref into a raw pointer then converting the raw pointer into jlong
+        let column_family_ptr: jlong = jlong::from(
```

Comment:
can we use `create_java_object` utility method here?

Reply:
The logic to create `ColumnFamily` Java-object is slightly different here - the address of a CF Rust-object residing inside of the Rust-instance of `Storage` is directly passed on the Java side. So the `ColumnFamily` Rust-object here should remain on it's own place without being moved into the `Box` as it is done in the `create_java_object` method for other Rust-objects.
The reason behind this is that `ColumnFamily` can't be copied to create another instance of the same CF because `ColumnFamily` object actually is just a wrapper for a pointer to internal memory of DB; so existence of two copies of the same CF (i.e. two copies of the same pointer) causes that the same memory will be freed twice: when a CF copy is destroyed and when DB is closed (in the `ColumnFamily` `drop` method https://github.com/HorizenOfficial/rust-rocksdb/blob/e2e8834975cc9f2cef63ae85d7ea376e7cdc816d/src/column_family.rs#L79).